File: Module 3/3A Javascript/JavaScript Coding Challenge/Task9.js
// Return the number of vowels in a string

function countVowels(str) {
    // Convert the string to lowercase to handle both uppercase and lowercase letters
    str = str.toLowerCase();
    
    // Define a set of vowels
    const vowels = ['a', 'e', 'i', 'o', 'u'];
    
    // Initialize a counter for the vowels
    let count = 0;
    
    // Iterate through each character in the string
    for (let char of str) {
        // Check if the character is a vowel
        if (vowels.includes(char)) {
            count++;
        }
    }
    
    // Return the count of vowels
    return count;
}

// Sample usage
let inputString = "Hello, World!";
let vowelCount = countVowels(inputString);
console.log(`The number of vowels in "${inputString}" is:`, vowelCount);

File: Module 4/4C_Everyday_Go_To_Python/Python_Coding_Challenge.py/Task9.py
"""
Return the number of vowels in a string.
"""
def count_vowels(input_string):
    vowels = "aeiouAEIOU"  # List of vowels (both lowercase and uppercase)
    count = 0  # Initialize vowel count

    for char in input_string:
        if char in vowels:  # Check if character is a vowel
            count += 1

    return count

# Example usage
input_string = "Hello World!"
vowel_count = count_vowels(input_string)
print(f"Number of vowels in '{input_string}': {vowel_count}")
